File: core/src/com/ore/infinium/components/BlockComponent.java
package com.ore.infinium.components;

import com.artemis.Component;

/**
 * ***************************************************************************
 * Copyright (C) 2014 by Shaun Reich <sreich02@gmail.com>                    *
 * *
 * This program is free software; you can redistribute it and/or            *
 * modify it under the terms of the GNU General Public License as           *
 * published by the Free Software Foundation; either version 2 of           *
 * the License, or (at your option) any later version.                      *
 * *
 * This program is distributed in the hope that it will be useful,          *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of           *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            *
 * GNU General Public License for more details.                             *
 * *
 * You should have received a copy of the GNU General Public License        *
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.    *
 * ***************************************************************************
 */

/**
 * used for blocks when they are held in the inventory or dropped in the world.
 * they're not used for sending tile regions of course, that would be expensive.
 * only arrays are used there.
 */
public class BlockComponent extends Component {
    public byte blockType;

    /**
     * copy a component (similar to copy constructor)
     *
     * @param blockComponent
     *         component to copy from, into this instance
     */
    public void copyFrom(BlockComponent blockComponent) {
        blockType = blockComponent.blockType;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("blockComponent.blockType: ").append(blockType).append('\n');
        return builder.toString();
    }
}

File: core/src/com/ore/infinium/components/BlockComponent.kt
/**
MIT License

Copyright (c) 2016 Shaun Reich <sreich02@gmail.com>

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
 */

package com.ore.infinium.components

import com.artemis.Component
import com.ore.infinium.util.ExtendedComponent

/**
 * used for blocks when they are held in the inventory or dropped in the world.
 * they're not used for sending tile regions of course, that would be expensive.
 * only arrays are used there.
 */
class BlockComponent : Component(), ExtendedComponent<BlockComponent> {

    var blockType: Byte = 0

    override fun canCombineWith(other: BlockComponent) =
            this.blockType == other.blockType

    override fun copyFrom(other: BlockComponent) {
        this.blockType = other.blockType
    }
}
